task_manager/test_task.hh

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "task_manager.hh"
+
+class test_task : public task_manager::task::impl {
+public:
+    test_task(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_id parent_id = task_id::create_null_id()) 
+        : task_manager::task::impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), parent_id)
+    {}
+
+    future<> abort() override {
+        return make_ready_future<>();
+    }
+
+    task_manager::task_ptr create_child(task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_id parent_id = task_id::create_null_id()) const override {
+        auto task_impl_ptr = std::make_unique<test_task>(_module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), parent_id);
+        return seastar::make_lw_shared<task_manager::task>(std::move(task_impl_ptr));
```

Comment:
this would be a good pointer to add the child to the parent task children list, before returning the child's task_ptr